aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -100,10 +104,11 @@ protected void doRefresh() {
   protected void doCommit(TableMetadata base, TableMetadata metadata) {
     String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
     boolean exceptionThrown = true;
-    Table glueTable = getGlueTable();
-    checkMetadataLocation(glueTable, base);
-    Map<String, String> properties = prepareProperties(glueTable, newMetadataLocation);
     try {
+      lock(newMetadataLocation);
```

Comment:
Because this locks in the `try` block, the finally will be called when the lock fails. I think that's correct for cleaning up the metadata location, but the lock release will currently fail because the lock isn't held by this thread. That will cause the lock failure exception to get replaced by the unlock failure.

I think the solution is to not throw exceptions in release. I'll comment on that below.